src/Kiota.Builder/Writers/Python/CodeClassDeclarationWriter.cs

```diff
@@ -49,20 +47,40 @@ public override void WriteCodeElement(ClassDeclaration codeElement, LanguageWrit
         }
     }
 
-    private void WriteInternalImports(ClassDeclaration codeElement, LanguageWriter writer, CodeNamespace parentNameSpace) {
+    private static void WriteInternalImports(ClassDeclaration codeElement, LanguageWriter writer) {
         var internalImportSymbolsAndPaths = codeElement.Usings
                                                         .Where(x => !x.IsExternal)
-                                                        .Select(x => _relativeImportManager.GetRelativeImportPathForUsing(x, parentNameSpace))
+                                                        .Select(x => GetImportPathForUsing(x))
                                                         .GroupBy(x => x.Item3)
                                                         .Where(x => !string.IsNullOrEmpty(x.Key))
                                                         .OrderBy(x => x.Key);
         if(internalImportSymbolsAndPaths.Any()){
             foreach (var codeUsing in internalImportSymbolsAndPaths)
-                writer.WriteLine($"from {codeUsing.Key.ToSnakeCase()} import {codeUsing.Select(x => GetAliasedSymbol(x.Item1, x.Item2)).Distinct().OrderBy(x => x).Aggregate((x,y) => x + ", " + y)}");
+                foreach(var symbol in codeUsing.Select(x => GetAliasedSymbol(x.Item1, x.Item2)).Distinct().OrderBy(x => x))
+                    writer.WriteLine($"{symbol} = lazy_import('{codeUsing.Key.ToSnakeCase().Replace("._", ".")}.{symbol}')"); 
             writer.WriteLine();
         }
     }
     private static string GetAliasedSymbol(string symbol, string alias) {
         return string.IsNullOrEmpty(alias) ? symbol : $"{symbol} as {alias}";
     }
+
+        /// <summary>
+    /// Returns the import path for the given using and import context namespace.
+    /// </summary>
+    /// <param name="codeUsing">The using to import into the current namespace context</param>
+    /// <returns>The import symbol, it's alias if any and the import path</returns>
+    private static (string, string, string) GetImportPathForUsing(CodeUsing codeUsing)
+    {
+        if (codeUsing?.IsExternal ?? true)
```

Comment:
I believe line 52 already filters out such instances. Maybe we should consider dropping one of them.

Reply:
Thanks, resolved in https://github.com/microsoft/kiota/pull/2028/commits/f9b35fc7ab4f18083d38e436a09d3343e9439226